content/tutorials/how-to-send-l1-l2-transaction/10.index.md

```diff
@@ -0,0 +1,335 @@
+---
+title: Send a transaction from L1 to L2
+description: This how-to guide explains how to send a transaction from Ethereum that interacts with a contract deployed on ZKsync.
+---
+
+The [ZKsync Era smart contracts](https://github.com/matter-labs/era-contracts/tree/main/l1-contracts/contracts/zksync)
+allow a sender to request transactions on Ethereum (L1) and pass data to a contract deployed on ZKsync Era (L2).
+
+In this example we'll send a transaction to an L2 contract from L1 using `zksync-ethers`, which provides helper methods to simplify the process.
+
+## Common use cases
+
+Along with ZKsync Era's built-in censorship resistance that requires multi-layer interoperability,
+there are some common use cases that need L1 to L2 transaction functionality, such as:
+
+- Custom bridges.
+- Multi-layer governing smart contracts.
+
+## Requirements
+
+Before continue, make sure you have the following:
+
+- Destination contract ABI: in order to execute a transaction on a contract on L2,
+  you'll need the contract's ABI. For this example we'll assume a `Greeter.sol` contract is deployed
+  on ZKsync ([like this one](https://sepolia.explorer.zksync.io/address/0x543A5fBE705d040EFD63D9095054558FB4498F88)).
+- L1 RPC endpoint: to broadcast the L1->L2 transaction to the network.
+  You can find [public Sepolia RPC endpoints in Chainlist](https://chainlist.org/chain/11155111).
+- Account with ETH in L1 to pay the gas fees. Use any of the [faucets listed here](https://docs.zksync.io/ecosystem/network-faucets#sepolia-faucets).
+
+## Set up
+
+1. Create a project folder and `cd` into it
+
+    ```sh
+    mkdir cross-chain-tx
+    cd cross-chain-tx
+    ```
+
+1. Initialise the project:
+
+    ::code-group
+
+    ```bash [npm]
+    npm init -y
+    ```
+
+    ```bash [yarn]
+    yarn init -y
+    ```
+
+    ::
+
+1. Install the required dependencies:
+
+    ::code-group
+
+    ```bash [npm]
+    npm install -D zksync-ethers ethers typescript dotenv @matterlabs/zksync-contracts @types/node
+    ```
+
+    ```bash [yarn]
+    yarn add -D zksync-ethers ethers typescript dotenv @matterlabs/zksync-contracts @types/node
+
+    ```
+
+    ::
+
+1. Install `ts-node` globally to execute the scripts that we're going to create:
+
+    ::code-group
+
+    ```bash [npm]
+    npm install -g ts-node
+    ```
+
+    ```bash [yarn]
+    yarn global add ts-node
+
+    ```
+
+    ::
+
+1. In the root folder, add a `.env` file with the private key of the wallet to use:
```

Comment:
should we add warning to only use test wallet in .env

Reply:
fixed